### ISSUE TITLE
Can you add Inverse Sigmoid too?

### ISSUE THREAD

**Issue by Corshi** [lodimasq/batch-checkpoint-merger#1]
According to this, there is something that it adds.
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/2658

But if not, I still just wanna tell you that this tool is amazing.
And thank you for making it.

I personally use it in my main python directory and it works great!

**Comment by Xyem** [comment]
I've thrown in a PR (#4) which implements the formula mentioned in the issue you linked. I'm not familiar with it enough to know if it is the correct one though. Hope it helps!

**Comment by lodimasq** [comment]
I'm glad you are enjoying the tool. The TLDR of below is that I'm not considering adding this right now. If enough people ask for it, I may reconsider but I wanted to at least give a reasonable explanation of why not.

The goal with implementing sigmoid-like funcs into the tool is to smooth out the transition between merge steps by artifically adjusting the distance between each step set by the user. 

Why do this in the first place? When models are merged at very low (close to 0) or very high (close to 1) ratios, small changes in the ratio have a much greater effect than ratios which are closer to the midpoint (0.5). The easiest way to explain this is to say the difference between 0.1-0.2 is much greater than the difference between 0.4-0.5.

The inverse of this function, does what you might expect it to do, the opposite. Inverse smoothstep will accelerate the rate of change between steps. You can see these effects by observing the differences between the steps in the image of the linked post. With regular smoothstep you can see that the character's rounder, more "anime" style face sticks around a lot longer than it does with inverse smoothstep. Another telling sign is how quickly the glitter appears on the dress, with inverse this is visible at only 20% where smoothstep this only appears around the 40% mark. 

Looking past the convergence point at 50%, it is much harder to tell in this example. However, look at the sharp transition between 90% and 100% with inverse smoothstep, there is no smooth transition here between dress and shorts. There is also no visible progression in the red waistband/ribbon detail.

The takeaway here? Inverse sigmoid/smoothstep is often times not doing anything helpful, in most cases it will make your adventure into model merging much more difficult. While subjectively, sometimes merges look better when a user ticks the box, this is nothing but chance. You are very unlikely to see a consistent improvement in results by using the inverse version of this function as it does the opposite of our goal from earlier.

When implementing smoothstep into this tool, I took careful note of why it was removed from Automatic111's repo, it's confusing for your typical user. I've tried (maybe not as sucessfully as I would like) to make this as clear as possible to the user what is happening when they enable it. This is the whole reason the graph exists in the tool in the first place. While I understand this might be a feature users want, I believe this comes from a place of ambiguity, where trial and error has lead to past sucessful results.

Sorry for the long reply, I wanted to be clear in my reasoning and I expect this question will be asked again in the future so this post can serve as a reference point.

**Comment by Corshi** [comment]
For your information, I did read it and I do appreciate the answer and time you took on it. Makes sense to me ;)

**Comment by Corshi** [comment]
Sorry for the post. I just figured things out, and had nowhere else to post it...

huh, I looked up what sigmoid really is. It not about being closer. It is mathematical subjectivity about movement and speed. In more layman's terms, it is how tempo of changes are nice for the eyes.
This merging is nothing to do with that, really. As „smoothstep” merging should be done with the function on the whole area of models. Instead this provides only a cut-out part in it (the function). The name should be "smoothstep's step". So to make a "smoothstep's step" of 0.25, I can just do a weighted sum of 0.1563.
To make a real  "smoothstep" merge of models, it would require to do merge of every model's layer, according to "smoothstep's steps". (or data segments it has)
So if one model had 1000layers and other had 1500 layers, it would only count those 1000 layers. So it would spread "smoothstep" into 1000 steps, to accommodate every layer (otherwise it would probably break –but I’m not sure about that). And any variation of "smoothstep" would be a variant of 50/50, as this is just a complete merge. The only reason for M in "smoothstep" would by to change central position. So M=0.25 would (in previous example) be counting first 250 layers as 250 to central point (so half "smoothstep" of 250) and 750 steps from central point (other half "smoothstep" by 750 steps). But that still would count as 50/50 of both models...
